File: src/main/java/com/ject/studytrip/stamp/domain/repository/StampQueryRepository.java
package com.ject.studytrip.stamp.domain.repository;

import com.ject.studytrip.stamp.domain.model.Stamp;
import java.util.List;
import java.util.Optional;

public interface StampQueryRepository {
    List<Stamp> findStampsToShiftAfterOrder(Long tripId, int deletedOrder);

    Optional<Stamp> findFirstIncompleteStampByTripId(Long tripId);
}

File: src/main/kotlin/com/ject/studytrip/stamp/domain/repository/StampQueryRepository.kt
package com.ject.studytrip.stamp.domain.repository

import com.ject.studytrip.stamp.domain.model.Stamp
import java.util.Optional

interface StampQueryRepository {
    fun findStampsToShiftAfterOrder(
        tripId: Long,
        deletedOrder: Int,
    ): List<Stamp>

    fun findFirstIncompleteStampByTripId(tripId: Long): Optional<Stamp>

    fun findNextStampOrderByTripId(tripId: Long): Int
}
